python/pyspark/shell.py

```diff
@@ -100,10 +100,11 @@
     % (platform.python_version(), platform.python_build()[0], platform.python_build()[1])
 )
 if is_remote():
-    print(
-        "Client connected to the Spark Connect server at %s"
-        % urlparse(os.environ["SPARK_REMOTE"]).netloc
-    )
+    url = os.environ.get("SPARK_REMOTE", None)
+    assert url is not None
+    if url.startswith("local"):
```

Comment:
Previously it was `SPARK_REMOTE` was replaced to `sc://localhost` via Spark session initialization. We don't do this anymore after https://github.com/apache/spark/pull/41013, and that's why it needs some manual change here for display.